### PR TITLE
Add NFRs for `DeveloperGuide` Page

### PR DESCRIPTION
**Pull request by Santosh3007**
## Changes
- Update and add non-functional requirements in developer guide

## Issues
- Address #6

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -402,10 +402,10 @@ Priorities:
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+2.  Should be able to hold up to 1000 events and contacts without a noticeable sluggishness in performance for typical usage.
```
Comment (ShenyiCui) [inline docs/DeveloperGuide.md:405]: `events` or `tasks`?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -402,10 +402,10 @@ Priorities:
...
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-*{More to be added}*
+4.  Data should be persistent and stored in the local machine's storage
```
Comment (ShenyiCui) [inline docs/DeveloperGuide.md:407]: spacing inconsistency

**Review by ShenyiCui — APPROVED** [review]
LGTm! Just a few minor changes

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (peppapighs) [reply docs/DeveloperGuide.md:405]: tasks

Comment (Santosh3007) [reply docs/DeveloperGuide.md:405]: Changed! :sparkles:

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Santosh3007) [reply docs/DeveloperGuide.md:407]: Resolved! 👍

**Pull request merged**